WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginBrowserActivity.java

```diff
@@ -296,8 +296,8 @@ public boolean onQueryTextSubmit(String query) {
     }
 
     @Override
-    public boolean onQueryTextChange(String query) {
-        mViewModel.setSearchQuery(query != null ? query : "");
+    public boolean onQueryTextChange(@NonNull String newText) {
+        mViewModel.setSearchQuery(newText);
```

Comment:
Similar to the comment about `afterTextChanged`, I wonder whether this is safe to do? I'm especially curious because the `query != null` ternary guard falls back to an empty string 🤔  - if this has survived a decade in production, do we risk introducing a "fail-fast crash" to replace the empty string fallback? I.e. is the existing implementation already more robust?

Reply:
Yea, I know, me looking at the change (05f5add71f6d70abcf8c8ec3a04289246999d9a1) that introduced this didn't give me enough evidence of its need, thus me being more bold on that, and again me basing this change on all other usages of it across the whole codebase (Java and Kotlin included, and especially Kotlin, where it is solely used as non-null, without the extra `?`). 🤞 

Having said that, and as per all the discussion we are having on this, "fail fast" vs "being safe", I wonder what is best, to keep having something there that might never be true, thus us needed to create a fallback, but then risking having every other client code that using this `onQueryTextChange(...)` think it being potentially falsely `@Nullable`, or risk a potential very low chance of a quickly revertible crash, especially since testing verified that this is indeed an empty string (`""`) instead of it being `null` when the search result is empty... 🤷 

PS: I personally think that we can just quickly smoke test this and all those cases and be better of having a `@NonNull` annotation, then spread that across the codebase, than have a `@Nullable` one and spread that. No matter, we should either should one or the other, I wouldn't be comfortable having a mix of those and thus ending up having an inconsistent signature for other to replicate inconsistently as well. Wdyt? 🤔

Reply:
The evidence you mention is compelling, imo, i.e. that this is _already_ being treated as `@NonNull` throughout the codebase and without issue. So, I agree with your rationale for consistently annotating it as such. That said, like the others, I think this could be on another PR with dedicated crash monitoring. My biggest concern is that we introduce a _new crash_ and negatively affect the user experience. If we are very confident we can avoid that (from the recognition of wide usage already, along with scrutiny during beta testing), I think it's ok to annotate as `@NonNull`.

Reply:
I agree on having a separate PR @mkevins ! 👍